src/SerializeAST.cc

```diff
@@ -351,6 +351,26 @@ sos::Object WrapNamedType(const mson::NamedType& namedType)
     return namedTypeObject;
 }
 
+sos::String WrapElementRole(const Element::Role& role)
+{
+    std::string str;
+
+    switch (role) {
+        case Element::BodyExampleRole:
+            str = "bodyExample";
+            break;
+
+        case Element::BodySchemaRole:
+            str = "bodySchema";
+            break;
+
+        default:
```

Comment:
I'd suggest adding `UndefinedRole` and removing default. That way if a new enumeration value is added we will get a compiler warning until it is explicitly handled, instead of it being caught by the default and potentially going missed when adding new enumeration values in the *distant* future.